core/receipt/datamodel/receipt.go

```diff
@@ -37,8 +37,8 @@ type MetaModel struct {
 }
 
 type ResultModel[O any, X any] struct {
-	Ok  O
-	Err X
+	Ok  *O
+	Err *X
```

Comment:
So, a union type per bindnode encoding is ALWAYS a pointer type. But the point just encodes whether or not it's present, which is then used to generate the keyed union. You'll note over in go-w3up you're forced to pass pointer types to the receipt. In fact, if you don't the whole code will panic. In my opinion, since no receipt that doesn't have a pointer type will EVER decode or encode properly, it perhaps makes better sense to hide the pointer requirement, which is IPLD specific, from the user.